BudgetSquirrel.Api/ClientApp/src/app/budgets/budget/budget.component.ts

```diff
@@ -0,0 +1,167 @@
+import { Component, OnInit, Input, Output, EventEmitter } from '@angular/core';
+import { Budget } from '../models';
+import { BudgetService } from '../services/budget.service';
+
+export interface EditBudgetEvent {
+  budget: Budget;
+  field: string;
+  value: string;
+}
+
+@Component({
+  selector: 'bs-budget',
+  template: `
+    <div class="budget-item__container budget-item--level-{{this.level}}">
```

Comment:
Can be done later but we should pull this html into its own file given it size or try to break it down into smaller components. Either way this can be hard to maintain over time.

Reply:
Yeah I agree. Maybe the next time we change this file, we also pull it out into the template?

Reply:
Sounds good.